nvflare/fuel/hci/zip_utils.py

```diff
@@ -36,29 +36,37 @@ def _get_default_meta(job_folder_name: str) -> str:
     return meta
 
 
-def _path_join(base: str, *parts: str) -> str:
-    path = os.path.normpath(os.path.join(base, *parts))
+def normpath_for_zip(path):
+    """Normalizes the path for zip file.
+
+    Args:
+        path (str): the path to be normalized
+    """
+    path = os.path.normpath(path)
     path = os.path.splitdrive(path)[1]
     # ZIP spec requires forward slashes
     return path.replace("\\", "/")
 
 
 def remove_leading_dotdot(path: str) -> str:
-    while path.startswith("../"):
+    while path.startswith(f"..{os.sep}"):
         path = path[3:]
     return path
 
 
 def split_path(path: str) -> (str, str):
-    """Split a path into prefix and folder name
+    """Splits a path into a pair of head and tail.
+
+    It removes trailing `os.sep` and call `os.path.split`
 
     Args:
         path: Path to split
 
-    Returns: A tuple of (prefix, folder_name)
+    Returns:
+        A tuple of `(head, tail)`
     """
 
-    if path.endswith("/"):
+    if path.endswith(os.sep):
```

Comment:
Should check for both slashes (forward and backward).

Reply:
use `path = str(Path(path))` first to unify forward and backward slashes in Windows.